dex/networks/eth/txdata.go

```diff
@@ -55,33 +149,46 @@ func ParseInitiateData(calldata []byte) ([]swapv0.ETHSwapInitiation, error) {
 		return nil, fmt.Errorf("expected first arg of type []swapv0.ETHSwapInitiation but got %T", args[0].value)
 	}
 
-	toReturn := make([]swapv0.ETHSwapInitiation, 0, len(initiations))
+	// This is done for the compiler to ensure that the type defined above and
+	// swapv0.ETHSwapInitiation are the same, other than the tags.
+	if len(initiations) > 0 {
+		_ = swapv0.ETHSwapInitiation(initiations[0])
+	}
+
+	toReturn := make([]*Initiation, 0, len(initiations))
 	for _, init := range initiations {
-		toReturn = append(toReturn, swapv0.ETHSwapInitiation(init))
+		gweiValue, err := ToGwei(init.Value)
+		if err != nil {
+			return nil, fmt.Errorf("cannot convert wei to gwei: %w", err)
+		}
+
+		toReturn = append(toReturn, &Initiation{
+			LockTime:    time.Unix(init.RefundTimestamp.Int64(), 0),
+			SecretHash:  init.SecretHash,
+			Participant: init.Participant,
+			Value:       gweiValue,
+		})
 	}
 
 	return toReturn, nil
 }
 
-// ParseRedeemData accepts call data from a transaction that pays to a
-// contract with extra data. It will error if the call data does not call
-// redeem with expected argument types. It returns the secret and secret hash
-// in that order.
-func ParseRedeemData(calldata []byte) ([]swapv0.ETHSwapRedemption, error) {
+func (t *txDataHandlerV0) parseRedeemData(calldata []byte) ([]*Redemption, error) {
 	decoded, err := parseCallData(calldata, swapv0.ETHSwapABI)
 	if err != nil {
 		return nil, fmt.Errorf("unable to parse call data: %v", err)
 	}
-	if decoded.name != redeemFuncName {
-		return nil, fmt.Errorf("expected %v function but got %v", redeemFuncName, decoded.name)
+	if decoded.name != t.redeemFuncName {
+		return nil, fmt.Errorf("expected %v function but got %v", t.redeemFuncName, decoded.name)
 	}
 	args := decoded.inputs
 	// Any difference in number of args and types than what we expect
 	// should be caught by parseCallData, but checking again anyway.
 	//
 	// TODO: If any of the checks prove redundant, remove them.
-	if len(args) != numRedeemArgs {
-		return nil, fmt.Errorf("expected %v redeem args but got %v", numRedeemArgs, len(args))
+	numArgs := 1
+	if len(args) != numArgs {
+		return nil, fmt.Errorf("expected %v redeem args but got %v", numArgs, len(args))
```

Comment:
I don't know that this is an improvement, but if you're going to do this, do a `const numArgs = 1`

Reply:
It was my opinion that having these integers as fields of these types made no sense given they can only ever be one specific value because the type is for a certain contract version.  `const` would be better though.

Reply:
Oh. I guess I was thinking that it might as well just be `if len(args) != 1`, but I wasn't being very clear. With `const`s, it's ok.